text/0020-bech32-address-format/0020-bech32-address-format.md

```diff
@@ -0,0 +1,78 @@
++ Feature name: `bech32-address-format`
++ Start date: 2020-07-28
++ RFC PR: [iotaledger/protocol-rfcs#0020](https://github.com/iotaledger/protocol-rfcs/pull/20)
+
+# Summary
+
+This document proposes an extendable address format for the IOTA protocol supporting various signature schemes and address types. It relies on the [Bech32](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki) format to provide a compact, human-readable encoding with strong error correction guarantees.
+
+# Motivation
+
+With [Chrysalis](https://roadmap.iota.org/chrysalis), IOTA uses Ed25519 to generate digital signatures, in which addresses correspond to a BLAKE2b-256 hash. It is necessary to define a new universal and extendable address format capable of encoding different types of addresses.
+
+The current IOTA protocol relies on Base27 addresses with a truncated Kerl checksum. However, both the character set and the checksum algorithm have limitations: 
+- Base27 is designed for ternary and is ill-suited for binary data.
+- The Kerl hash function also requires ternary input. Further, it is slow and provides no error-detection guarantees.
+- It does not support the addition of version or type information to distinguish between different kinds of addresses with the same length.
+
+All of these points are addressed in the Bech32 format introduced in [BIP-0173](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki): In addition to the usage of the human-friendly Base32 encoding with an optimized character set, it implements a [BCH code](https://en.wikipedia.org/wiki/BCH_code) that _guarantees detection_ of any error affecting at most four characters and has less than a 1 in 10<sup>9</sup> chance of failing to detect more errors.
+
+This RFC proposes a simple and extendable binary serialization for addresses of different types that is then Bech32 encoded to provide a unique appearance for human-facing applications such as wallets. 
+
+# Detailed design
+
+## Binary serialization
+
+The address format uses a simple serialization scheme which consists of two parts:
+
+   - The first byte describes the type of the address.
+   - The remaining bytes contain the type-specific raw address bytes.
+
+Currently, only one kind of addresses are supported:
+ - Ed25519, where the address consists of the BLAKE2b-256 hash of the Ed25519 public key.
+
+They are serialized as follows:
+
+| Type    | First byte | Address bytes                                             |
+| ------- | ---------- | --------------------------------------------------------- |
+| Ed25519 | `0x01`     | 32 bytes: The BLAKE2b-256 hash of the Ed25519 public key. |
```

Comment:
The first and only supported address type (Ed25519) uses `1` instead of `0`. This tries to stay consistent with the address type in other RFCs, but should be unified once those types have been finalized.

Reply:
Should we just go forward, remove WOTS from the other RFCs and use 0 ?